### PR TITLE
Expose props to CatalogGraphPage

### PR DESCRIPTION
**Pull request by shubhamjajoo**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by shubhamjajoo** [comment]
@benjdlambert   Updated the changeset. but still it's failing

**Comment by shubhamjajoo** [comment]
@benjdlambert can you pls review ?

**Review by benjdlambert — APPROVED** [review]
Thanks for the PR! Looks like you've found an issue with the types going to raise a PR to fix that now, and then we can rebase this PR on top of that.

**Comment by shubhamjajoo** [comment]
@benjdlambert , want to track progress for same. Need this feature for CatalogGraphPage at earliest. 🙂

**Comment by benjdlambert** [comment]
@shubhamjajoo I think that the original issue has been fixed for those types now, do you want to try and rebase this? :praY:

**Comment by benjdlambert** [comment]
https://github.com/backstage/backstage/pull/17427

**Comment by shubhamjajoo** [comment]
@benjdlambert rebase my branch. 
PTAL once.

File: plugins/catalog-graph/src/components/CatalogGraphCard/CatalogGraphCard.tsx [hunk 1]
```diff
@@ -139,6 +141,8 @@ export const CatalogGraphCard = (
         mergeRelations={mergeRelations}
         direction={direction}
         relationPairs={relationPairs}
+        renderLabel={renderLabel}
```
Comment (sennyeya) [inline plugins/catalog-graph/src/components/CatalogGraphCard/CatalogGraphCard.tsx:144]: These should be collapsed into the `{...props}` above, we don't need to add them again here. The other options are defined here because there are default values provided either in lines 70-18 or in code before this block.

File: plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx [hunk 2]
```diff
@@ -105,6 +107,8 @@ const useStyles = makeStyles(
 
 export const CatalogGraphPage = (props: {
   relationPairs?: RelationPairs;
+  renderNode?: Types.RenderNodeFunction<EntityNode>;
```
Comment (sennyeya) [inline plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx:110]: Can we add these instead as 
```
graphProps: Partial<EntityRelationsGraphProps>
```
similar to the CatalogGraphCard work? You can then pass it as `{...props}` to the EntityRelationsGraph and it allow for more than just the two props.

As I mentioned in my comment on the issue, I think there's also some work needed to shift some initialState into that prop, but it shouldn't be part of this PR.

**Review by sennyeya — CHANGES_REQUESTED** [review]
Looks good. Small feedback to support additional props that may come up in the future for the component.

File: plugins/catalog-graph/src/components/CatalogGraphCard/CatalogGraphCard.tsx (hunk 1 above)
Comment (shubhamjajoo) [reply plugins/catalog-graph/src/components/CatalogGraphCard/CatalogGraphCard.tsx]: @sennyeya  Updated. PTAL.

File: plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx (hunk 2 above)
Comment (shubhamjajoo) [reply plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx]: @sennyeya  Updated. PTAL.

Comment (shubhamjajoo) [reply plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx]: @sennyeya Once we merge this PR, maybe we can create one more issue for moving initialState into prop.
Happy to contribute on same.

File: .changeset/hungry-foxes-divide.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-graph': patch
+---
+
+Expose props `renderNode` & `renderLabel` to Catalog Graph Page
```
Comment (sennyeya) [inline .changeset/hungry-foxes-divide.md:5]: We should update this to include the inclusion of all props from the underlying graph not just these 2.

File: plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx [hunk 4]
```diff
@@ -103,20 +104,22 @@ const useStyles = makeStyles(
...
-    unidirectional?: boolean;
-    mergeRelations?: boolean;
-    direction?: Direction;
-    showFilters?: boolean;
-    curve?: 'curveStepBefore' | 'curveMonotoneX';
-  };
-}) => {
+export const CatalogGraphPage = (
+  props: {
+    relationPairs?: RelationPairs;
```
Comment (sennyeya) [inline plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx:109]: nit: Can we remove this as it's exposed from the EntityRelationsGraphProps already?

**Review by sennyeya — CHANGES_REQUESTED** [review]
A few more comments. Thanks for the quick turnaround on this, I'm surprised how terse it became by switching to `Partial<EntityRelationsGraphProps>`.

File: plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx (hunk 4 above)
Comment (shubhamjajoo) [reply plugins/catalog-graph/src/components/CatalogGraphPage/CatalogGraphPage.tsx]: Done

**Comment by shubhamjajoo** [comment]
@benjdlambert can you pls review and merge this?

**Comment by benjdlambert** [comment]
Thanks
